applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2016, prpl Foundation
+//
+// Permission to use, copy, modify, and/or distribute this software for any purpose with or without
+// fee is hereby granted, provided that the above copyright notice and this permission notice appear
+// in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE
+// INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE
+// FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+// ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+//
+// Author: Nils Koenig <openwrt@newk.it>
+// JS version: Ramon van Gorkom 
+
+
+'use strict';
+'require view';
+'require form';
+'require fs';
+'require uci';
+
+function timeValidator(value, desc) {
+	if (value !== null) {
+		const matches = value.match(/^(\d\d?):(\d\d?)$/);
+		if (matches) {
+			const h = parseInt(matches[1], 10);
+			const m = parseInt(matches[2], 10);
+			if (h >= 0 && h <= 23 && m >= 0 && m <= 59) {
+				return true;
+			}
+		}
+	}
+	return _('The value %s is invalid'.format(desc));
+}
+
+function checkFile(path) {
+	try { 
+		if (fs.stat(path)) { 
+			return true; 
+		} 
+	} catch (err) { 
+		return false;
+	}
+}
+
+return view.extend({
+
+	render: function () {
+		var m, s, o, oUnloadmodules, oModules;
+
+		m = new form.Map('wifi_schedule', _('Wifi Schedule'),_('Defines a schedule when to turn on and off wifi.'));
+
+		s = m.section(form.TypedSection, 'global', _('Manual control'));
+		s.optional = false;
+		s.rmempty = false;
+		s.anonymous = true;
+
+		o = s.option(form.Button, '', _('Activate wifi'));
+		o.write = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['start', 'manual']);
+		}
+
+
+		o = s.option(form.Button, '', _('Disable wifi gracefully'));
+		o.write = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['stop', 'manual']);
+		}
+
+		o = s.option(form.Button, '', _('Disable wifi forced'));
+		o.write = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['forcestop', 'manual']);
+		}
+
+
+		s = m.section(form.TypedSection, 'global', _('Global Settings'));
+		s.optional = false;
+		s.rmempty = false;
+		s.anonymous = true;
+
+
+
+		o = s.option(form.Flag, 'enabled', _('Enable Wifi Schedule'));
+		o.optional = false;
+		o.rmempty = false;
+		o.validate = function(section_id, value) {
+			if (value == '1') {
+				if ((checkFile('/sbin/wifi')) && 
+					(checkFile('/usr/bin/wifi_schedule.sh'))) {
```

Comment:
This is OK but I'm thinking we can perform this check in a Promise at load time. SOP for many modules.

Reply:
Let me check how other aps do it!